Controllers/ClanController.cs

```diff
@@ -286,8 +315,41 @@ public async Task<ActionResult> DeleteClan([FromQuery] int? id = null)
                 return NotFound();
             }
 
+            // Recalculate clanRanking on leaderboards that have this clan in their clanRanking
+            var leaderboardsRecalc = _context
+                .Leaderboards
+                .Include(lb => lb.ClanRanking)
+                .ThenInclude(cr => cr.AssociatedScores)
+                .Where(lb => lb.ClanRanking != null ? 
+                lb.ClanRanking.Any(lbClan => lbClan.Clan.Tag == clan.Tag) && lb.Difficulty.Status == DifficultyStatus.ranked : 
+                lb.Difficulty.Status == DifficultyStatus.ranked)
+                .ToList();
+
+            // Remove the clanRankings
+            leaderboardsRecalc.ForEach(leaderboard =>
+            {
+                var crToRemove =
+                    leaderboard.ClanRanking?
+                    .Where(cr => cr.Clan == clan)
+                    .FirstOrDefault();
+                if (crToRemove != null)
+                {
+                    // Do we need to remove the clanRanking from both the leaderboard and the ClanRanking table?
```

Comment:
You can remove just from the ClanRanking as you have cascade delete for the reference

Reply:
I removed the comment. With cascade deletion, do I still have to do the following line? 
crToRemove.AssociatedScores?.Clear();
I remember getting crashes without that line, but reading the docs, it feels like Associated Scores should be unlinked from ClanRanking if the clanRanking is removed.